test/functional/hosts_controller_test.rb

```diff
@@ -368,6 +368,17 @@ def setup_multiple_environments
     assert Host.find(@host2.id).environment == environments(:global_puppetmaster)
   end
 
+  test "user with edit host rights with update environments should change owner" do
+    @request.env['HTTP_REFERER'] = hosts_path
+    assert_equal users(:admin).id_and_type, @host1.is_owned_by
```

Comment:
`@host1` and 2 are populated by the `setup_multiple_environments` helper.  I'd suggest just putting the FG.create_list call (minus `:environment`) into your test.